src/routes/index.tsx

```diff
@@ -34,33 +29,13 @@ const SafeContainer = React.lazy(() => import('./safe/container'))
 
 const Routes = (): React.ReactElement => {
   const location = useLocation()
-  const { pathname, search } = location
   const defaultSafe = useSelector(lastViewedSafe)
-  const { trackPage } = useAnalytics()
 
   // Component key that changes when addressed route slug changes
   const { key } = useAddressedRouteKey()
 
-  // Google Analytics
-  useEffect(() => {
-    let trackedPath = pathname
-
-    // Anonymize safe address
-    if (hasPrefixedSafeAddressInUrl()) {
-      trackedPath = trackedPath.replace(getPrefixedSafeAddressSlug(), 'SAFE_ADDRESS')
-    }
-
-    // Anonymize deeplinked transaction
-    if (isDeeplinkedTx()) {
-      const match = matchPath(pathname, {
-        path: SAFE_ROUTES.TRANSACTIONS_SINGULAR,
-      })
-
-      trackedPath = trackedPath.replace(match?.params[TRANSACTION_ID_SLUG], 'TRANSACTION_ID')
-    }
-
-    trackPage(trackedPath + search)
-  }, [pathname, search, trackPage])
+  // Google Tag Manager page tracking
+  useGTMPageTracking()
```

Comment:
Why are we calling `unsubscribe` only when this component unmounts instead of when `pathname` changes like before?

Reply:
The listener handles any `location` changes internally. We need not worry about that anymore.